docs/source/conf.py

```diff
@@ -63,6 +63,7 @@
     "--InlineBackend.rc={'figure.dpi': 96}",
 ]
 
+nbsphinx_allow_errors = True
```

Comment:
This would remove much of the value of CI, as broken notebooks would fail silently.

Reply:
once I restored your raises-exception in the correct place, this was no longer necessary. removed this in one of the new commits below.